### PR TITLE
improve error handling

### PR DESCRIPTION
**Pull request by alexgomezlf**
- move error handling to top level
- show error to user when applicable

### CODE REVIEW

File: src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx [hunk 1]
```diff
@@ -91,7 +91,8 @@ export class SaveDocumentToLaserfiche {
       const spFileDataBlob = await res.blob();
       return spFileDataBlob;
     } catch (error) {
-      console.log('error ocurred' + error);
+      window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
```
Comment (rfulton-lf) [inline src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx:94]: remove the try/catch here

File: src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx [hunk 2]
```diff
@@ -386,13 +391,10 @@ export class SaveDocumentToLaserfiche {
...
     };
     const response = await fetch(fullSpFileUrl, init);
-    if (response.ok) {
-      window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
-      //Perform further activity upon success, like displaying a notification
-      alert('File deleted successfully');
-    } else {
+    window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
+    if (!response.ok) {
       window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
```
Comment (rfulton-lf) [inline src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx]: duplicate window.localStorage.removeItem

File: src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx [hunk 3]
```diff
@@ -415,11 +417,13 @@ export class SaveDocumentToLaserfiche {
       },
     });
     if (deleteFile.ok) {
-      alert('File replaced with link successfully');
       await this.replaceFileWithLinkAsync(filenameWithoutExt, docFilelink);
-    } else {
+    }
+    if (!deleteFile.ok) {
       window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
```
Comment (rfulton-lf) [inline src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx:416]: check the removeItem here

File: src/extensions/savetoLaserfiche/SaveToLaserficheDialog.tsx [hunk 4]
```diff
@@ -77,33 +81,46 @@ function SaveToLaserficheDialog(props: {
...
-          // TODO is this handled correctly when an error occured when deleting the file
-          props.successSave(false);
-          await props.closeClick();
+        try {
+          const successSaveToLF =
+            await saveToLF.trySaveDocumentToLaserficheAsync();
+          if (successSaveToLF) {
+            props.isSuccessfulLoggedIn(true);
+            setSuccess(successSaveToLF);
+          } else {
```
Comment (rfulton-lf) [inline src/extensions/savetoLaserfiche/SaveToLaserficheDialog.tsx]: check if else block is reachable or if error throws

File: src/webparts/LaserficheRepositoryAccessWebPart/components/RepositoryViewWebPart.tsx [hunk 5]
```diff
@@ -376,18 +376,20 @@ function ImportFileModal(props: {
...
-      lfFieldsService = new LfFieldsService(props.repoClient);
-      await fieldContainer.current.initAsync(lfFieldsService);
+      try {
+        fieldContainer.current.addEventListener('dialogOpened', onDialogOpened);
+        fieldContainer.current.addEventListener('dialogClosed', onDialogClosed);
+
+        lfFieldsService = new LfFieldsService(props.repoClient);
+        await fieldContainer.current.initAsync(lfFieldsService);
+      } catch (err) {
+        setFileUploadPercentage(0);
```
Comment (rfulton-lf) [inline src/webparts/LaserficheRepositoryAccessWebPart/components/RepositoryViewWebPart.tsx]: do we need to show to the user

File: src/webparts/LaserficheRepositoryAccessWebPart/components/RepositoryViewWebPart.tsx [hunk 6]
```diff
@@ -396,88 +398,94 @@ function ImportFileModal(props: {
...
-      } catch (error) {
-        window.alert('Error uploding file:' + JSON.stringify(error));
+      } else {
+        fieldContainer.current.forceValidation();
       }
-    } else {
-      fieldContainer.current.forceValidation();
+    } catch (err) {
+      setFileUploadPercentage(0);
+      setError(err.message);
```
Comment (rfulton-lf) [inline src/webparts/LaserficheRepositoryAccessWebPart/components/RepositoryViewWebPart.tsx:485]: check to see if we need to log errors elsewhere

File: src/webparts/laserficheAdminConfiguration/components/AddNewManageConfiguration/AddNewManageConfiguration.tsx [hunk 7]
```diff
@@ -127,38 +125,34 @@ export default function AddNewManageConfiguration(
...
-        return suceeded;
+        await saveNewPageConfigurationAsync();
       }
+    } else {
+      throw Error('Invalid configuration. Please review any errors.');
     }
-    return false;
   }
 
   async function GetItemIdByTitle(): Promise<IListItem[]> {
```
Comment (rfulton-lf) [inline src/webparts/laserficheAdminConfiguration/components/AddNewManageConfiguration/AddNewManageConfiguration.tsx]: consider renaming this function to be more descriptive

File: src/webparts/laserficheAdminConfiguration/components/ManageConfigurationsPage/ManageConfigurationsPage.tsx [hunk 8]
```diff
@@ -29,20 +28,21 @@ export default function ManageConfigurationsPage(
...
-      if (configurations?.configs.length > 0) {
-        setConfigRows(configRows.concat(...configurations.configs));
+      try {
+        const configurations: { id: string; configs: ProfileConfiguration[] } =
+          await getManageConfigurationsAsync();
+        if (configurations?.configs.length > 0) {
+          setConfigRows(configRows.concat(...configurations.configs));
+        }
+      } catch (err) {
+        console.warn(`Error: ${err.message}`);
```
Comment (rfulton-lf) [inline src/webparts/laserficheAdminConfiguration/components/ManageConfigurationsPage/ManageConfigurationsPage.tsx]: the console.warn's should probably all be console.error's

File: src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx [hunk 9]
```diff
@@ -45,20 +44,20 @@ export default function ManageMappingsPage(
...
+    try {
+      await getAllSharePointContentTypesAsync();
+      await getAllLaserficheContentTypesAsync();
+      const results: { id: string; mappings: ProfileMappingConfiguration[] } =
+        await getManageMappingsAsync();
+      if (results?.mappings.length > 0) {
+        setMappingRows(mappingRows.concat(results.mappings));
+      }
+    } catch (err) {
+      console.warn(`Error getting mappings: ${err.message}`);
```
Comment (rfulton-lf) [inline src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx]: -> console.error preferably

File: src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx (hunk 1 above)
Comment (alexgomezlf) [reply src/extensions/savetoLaserfiche/SaveDocumentToLaserfiche.tsx:94]: checked and it would be unlikely to reach here but should remove if it does

**Review by rfulton-lf — APPROVED** [review]

**Pull request merged**